vms/platformvm/txs/fee/calculator.go

```diff
@@ -11,105 +11,123 @@ import (
 	"github.com/ava-labs/avalanchego/vms/platformvm/upgrade"
 )
 
-var _ txs.Visitor = (*Calculator)(nil)
+var _ txs.Visitor = (*calculator)(nil)
+
+func NewStaticCalculator(config StaticConfig, upgradeTimes upgrade.Config) Calculator {
+	return Calculator{
+		config:       config,
+		upgradeTimes: upgradeTimes,
+	}
+}
 
 type Calculator struct {
+	config       StaticConfig
+	upgradeTimes upgrade.Config
+}
+
+func (c Calculator) GetFee(tx txs.UnsignedTx, time time.Time) uint64 {
```

Comment:
nit: I'd rename this `ComputeFee` to reduce forward differences.
The reason is that with dynamic fees I will need to introduce a GetFee to access calculator.fee multiple times, while a tx is being built and calculator.fee gets updated.